lib/api/filter.rb

```diff
@@ -120,24 +122,16 @@ def single_expression(field, operator, value)
     end
 
     def split_filter_string(filter)
-      operator = nil
-      operators_from_longest_to_shortest = OPERATORS.keys.sort_by(&:size).reverse
-      filter.size.times do |i|
-        operator = operators_from_longest_to_shortest.detect do |o|
-          o == filter[(i..(i + o.size - 1))]
-        end
-        break if operator
-      end
+      filter_match = OPERATOR_REGEXP.match(filter)
+      filter_attr, operator, filter_value = filter_match[1..3] if filter_match
```

Comment:
```suggestion
      filter_attr, operator, filter_value = filter_match.captures if filter_match
```

Reply:
Alternatively it might be nice to use named captures in the original regex with:

```ruby
    OPERATOR_REGEXP = /^ *(?<filter_attr>[^=!>< ]+) *(?<operator>!==?|[=!]~|[<>=]=?) *(?<filter_value>.*?) *$/.freeze
```

Then here you can use `values_at` or `named_captures` (or continue to just use `captures`)

Reply:
I tried a few times. was having getting it to find the captures. may revisit at a later time.